readthedocs/search/tests/conftest.py

```diff
@@ -16,27 +17,17 @@ def mock_elastic_index(mocker):
     mocker.patch.object(Index, '_index', index_name.lower())
 
 
-@pytest.fixture(autouse=True)
-def es_index(mock_elastic_index):
-    # Create the index.
-    index = Index()
-    index_name = index.timestamped_index()
-    index.create_index(index_name)
-    index.update_aliases(index_name)
-    # Update mapping
-    proj = ProjectIndex()
-    proj.put_mapping()
-    page = PageIndex()
-    page.put_mapping()
-    sec = SectionIndex()
-    sec.put_mapping()
-
-    yield index
-    index.delete_index(index_name=index_name)
+@pytest.fixture()
+def es_index():
+    call_command('search_index', '--delete', '-f')
+    call_command('search_index', '--create')
+
+    yield
+    call_command('search_index', '--delete', '-f')
 
 
 @pytest.fixture
-def all_projects():
+def all_projects(es_index):
```

Comment:
Where does this get passed in? Is it automatically callign the above fixture based on name?

Reply:
Actually, its pytest's [dependeny enjection](https://docs.pytest.org/en/latest/fixture.html#fixtures-a-prime-example-of-dependency-injection). So if you have a fixture name `foo` and you accept this fixture in `def bar(foo)`, the `foo` fixture will be passed to the `bar` fixture.